userspace/libscap/scap.c

```diff
@@ -323,7 +323,16 @@ scap_t* scap_open_live_int(char *error, int32_t *rc,
 		//
 		// Allocate the device descriptors.
 		//
-		len = RING_BUF_SIZE * 2;
+
+		FILE * fp = fopen("/sys/module/" PROBE_NAME "/parameters/ring_buf_size", "r");
+		if (fp == NULL){
+			snprintf(error, SCAP_LASTERR_SIZE, "Could not read module parameter ring_bug_size at '/sys/module/" PROBE_NAME "/parameters/ring_buf_size'");
```

Comment:
Would it be worth adding a fallback to the current default value? (In case libscap doesn't match the kernel module value)

Reply:
ring_**bug**_size